### PR TITLE
Disable cmdlets that are not supported under unix system

### LINKED ISSUE #4355: Determine *-PSSession and *-PSSessionConfiguration Cmdlet Support Via OMI
These cmdlets are included in non-Windows platforms, but are they supported?

``` powershell
Connect-PSSession
Disable-PSSessionConfiguration
Disconnect-PSSession
Enable-PSSessionConfiguration
Get-PSSessionCapability
Get-PSSessionConfiguration
New-PSSessionConfigurationFile
New-PSSessionOption
Receive-PSSession
Register-PSSessionConfiguration
Set-PSSessionConfiguration
Test-PSSessionConfigurationFile
Unregister-PSSessionConfiguration
```

If they are not supported, they should be removed from [here](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/InitialSessionState.cs#L5952)

Environment data
----------------
v6.0.0-beta.3

### PR DESCRIPTION
**Pull request by chunqingchen**
resolve #4355

Removed cmdlets from IntialSessionState.cs so they will not be available on unix platforms.

### CODE REVIEW

**Comment by iSazonov** [comment]
Usually we exclude not ported cmdlets with `#if !UNIX` (or exclude whole file in csproj) and remove from [psd1](https://github.com/PowerShell/PowerShell/tree/master/src/Modules/Unix)

Also we need correct tests in [DefaultCommands.Tests.ps1](https://github.com/PowerShell/PowerShell/blob/master/test/powershell/engine/Basic/DefaultCommands.Tests.ps1)

**Comment by chunqingchen** [comment]
@iSazonov in issue  #4355 you mentioned not implantation exception should be added instead of excluding the cmdlets. Is that not the case now?

**Comment by iSazonov** [comment]
@chunqingchen Sorry, my comment was not approved. I remember @daxian-dbw said that it is better exclude code in csproj - it is easy to track. @daxian-dbw Could you confirm?

File: test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1 [hunk 1]
```diff
@@ -280,3 +280,33 @@ Describe "Remoting loopback tests" -Tags @('CI', 'RequireAdminOnWindows') {
...
+
+Describe "Verify remoting cmdlets that should not be implemented under non-windows platform" -Tags @("CI") {
+    $testcases = @(
+                  @{command = "Connect-PSSession -Name name"; testname = "Connect-PSSession"}
+                  @{command = "Disable-PSSessionConfiguration"; testname = "Disable-PSSessionConfiguration"}
+                  @{command = "Disconnect-PSSession -Name name"; testname = "Disconnect-PSSession"}
+                  @{command = "Enable-PSSessionConfiguration"; testname = "Enable-PSSessionConfiguration"}
+                  @{command = "Get-PSSessionCapability name name"; testname = "Get-PSSessionCapability"}
+                  @{command = "Get-PSSessionConfiguration"; testname = "Get-PSSessionConfiguration"}
+                  @{command = "New-PSSessionConfigurationFile -Path c:\path"; testname = "New-PSSessionConfigurationFile"}
```
Comment (adityapatwardhan) [inline test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: For consistency sake, can we use non-windows path here?

File: test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1 [hunk 2]
```diff
@@ -280,3 +280,33 @@ Describe "Remoting loopback tests" -Tags @('CI', 'RequireAdminOnWindows') {
...
+                  @{command = "New-PSSessionOption"; testname = "New-PSSessionOption"}
+                  @{command = "Receive-PSSession -Name name"; testname = "Receive-PSSession"}
+                  @{command = "Register-PSSessionConfiguration -Name name"; testname = "Register-PSSessionConfiguration"}
+                  @{command = "Set-PSSessionConfiguration -Name name"; testname = "Set-PSSessionConfiguration"}
+                  @{command = "Test-PSSessionConfigurationFile -Path c:\path"; testname = "Test-PSSessionConfigurationFile"}
```
Comment (adityapatwardhan) [inline test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: For consistency sake, can we use non-windows path here?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by chunqingchen** [comment]
@adityapatwardhan your comments are resolved.

**Comment by TravisEz13** [comment]
The last commit is lot linked to a github account.   Please resolve.

**Comment by adityapatwardhan** [comment]
@chunqingchen The title and the description needs to be updated to match the implementation.

**Comment by chunqingchen** [comment]
@adityapatwardhan updated

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 3]
```diff
@@ -5587,4 +5587,4 @@ internal class RunspaceEventSource : EventSource
         public void LoadCommandStart(string Name) { WriteEvent(25, Name); }
         public void LoadCommandStop(string Name) { WriteEvent(26, Name); }
     }
-}
+}
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Please add new line at EOF.

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 4]
```diff
@@ -529,4 +529,4 @@ Describe "Verify approved aliases list" -Tags "CI" {
     It "Should have 'more' as a function" {
         Test-Path Function:more | Should Be $true
     }
-}
+}
```
Comment (iSazonov) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: Please add new line at EOF.

**Comment by chunqingchen** [comment]
@iSazonov your comment is resolved

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 (hunk 4 above)
Comment (iSazonov) [reply test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: @chunqingchen Sorry, we should add newline after the last brace not before.

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 3 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: @chunqingchen Sorry, we should add newline after the last brace not before.

**Review by iSazonov — COMMENTED** [review]
Leave a comment

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 5]
```diff
@@ -5587,4 +5587,5 @@ internal class RunspaceEventSource : EventSource
         public void LoadCommandStart(string Name) { WriteEvent(25, Name); }
         public void LoadCommandStop(string Name) { WriteEvent(26, Name); }
     }
-}
+
+}
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: missing newline

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 6]
```diff
@@ -529,4 +529,5 @@ Describe "Verify approved aliases list" -Tags "CI" {
     It "Should have 'more' as a function" {
         Test-Path Function:more | Should Be $true
     }
-}
+
+}
```
Comment (TravisEz13) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: missing newline

**Review by TravisEz13 — APPROVED** [review]
Fix minor issue and it should be good

**Comment by chunqingchen** [comment]
@TravisEz13 new line added.

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 7]
```diff
@@ -5587,4 +5587,5 @@ internal class RunspaceEventSource : EventSource
         public void LoadCommandStart(string Name) { WriteEvent(25, Name); }
         public void LoadCommandStop(string Name) { WriteEvent(26, Name); }
     }
+
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Please revert the formatting change and remove the line.

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 8]
```diff
@@ -529,4 +529,5 @@ Describe "Verify approved aliases list" -Tags "CI" {
     It "Should have 'more' as a function" {
         Test-Path Function:more | Should Be $true
     }
+
```
Comment (iSazonov) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: Please revert the formatting change and remove the line.

**Review by iSazonov — COMMENTED** [review]
Should we exclude the code from compilation?

**Comment by adityapatwardhan** [comment]
@chunqingchen Please run all tests using `[Feature]` in your git commit message.

https://github.com/PowerShell/PowerShell/blob/a1b7f8be3ea4ac7e4f7a67e6261ad68cdb23143d/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr

**Comment by chunqingchen** [comment]
@adityapatwardhan @iSazonov your comments are resolved. Thanks

**Comment by iSazonov** [comment]
I believe the PR need #5378 to pass CI Appveyor.

**Comment by adityapatwardhan** [comment]
@chunqingchen Can you have a look at the Travis CI failure?

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
CI Travis was passed but not reported.

**Comment by adityapatwardhan** [comment]
Merging as CI has passed but webhook did not update status to GitHub.

**Comment by vinodc** [comment]
@paulcallen Would you happen to know if `New-PSSessionOption` has been disabled? We updated to the latest version from 6.0.0 beta1 and this no longer works. It was previously working great and we have been using it with `New-PSSessionOption` since https://github.com/PowerShell/psl-omi-provider/issues/67 was merged in.

**Comment by dantraMSFT** [comment]
@vinodc I'll dig into this.  For now, you can use [System.Management.Automation.Remoting.PSSessionOption]::new() and populate the returned object.

BTW: What's the use case you have for it on Linux.  Perhaps it was missed with the cmdlet was removed.

**Comment by vinodc** [comment]
@dantraMSFT Thanks for the quick response. We are programmatically accessing Office 365 PowerShell cmdlets and in the process our code creates PSSessionOption objects. We can work around this, but wanted to make sure that the overall implementation itself is still supported for related commands used to connect to Office 365 PowerShell, starting with `New-PSSessionOption`.

On a broader note, this is related to piped commands failing (https://github.com/PowerShell/PowerShell/issues/6206), where we were recommended to update PowerShell first to see if we could reproduce the failure on the latest version, which resulted in these new issues coming up.

**Comment by dreyTee** [comment]
> @paulcallen Would you happen to know if `New-PSSessionOption` has been disabled? 
Have same issue when trying to  use WINRM from linux without proper server certificate:
Getting error:
`Use the PSSessionOption -SkipCACheck and -SkipCNCheck`
Trying to use thes options with `New-PSSessionOption` but can't:
`New-PSSessionOption : The term 'New-PSSessionOption' is not recognized as the name of a cmdlet, function, script file, or operable program.`
Please advise a workaround a least.

**Comment by iSazonov** [comment]
@dreyTee Please open new issue for your question.

**Pull request merged**